java/rocksjni/rocksjni.cc

```diff
@@ -1181,6 +1181,86 @@ jint Java_org_rocksdb_RocksDB_getDirect(JNIEnv* env, jobject /*jdb*/,
       jkey, jkey_off, jkey_len, jval, jval_off, jval_len, &has_exception);
 }
 
+jint rocksdb_get_unsafe_helper(
+    JNIEnv* env, ROCKSDB_NAMESPACE::DB* db,
+    const ROCKSDB_NAMESPACE::ReadOptions& read_options,
+    ROCKSDB_NAMESPACE::ColumnFamilyHandle* column_family_handle,
+    jbyteArray jkey, jint jkey_off, jint jkey_len, jlongArray jret_data,
+    bool* has_exception) {
+  static const int kNotFound = -1;
+  static const int kStatusError = -2;
+
+  char* key =
+      reinterpret_cast<char*>(env->GetPrimitiveArrayCritical(jkey, nullptr));
```

Comment:
use direct buffer to pass key that to prevent GetPrimitiveArrayCritical+ReleasePrimitiveArrayCritical?